.github/workflows/test.yml

```diff
@@ -1,52 +1,46 @@
 ---
-name: Test 📦
-
-on:
-  pull_request:
-    types:
-      - opened
-      - synchronize
-      - reopened
-      - ready_for_review
-    branches:
-      - main
-  push:
-    branches:
-      - main
+name: Test and QC 🧪
+
+on:  
   workflow_call:
 
 concurrency:
   group: test-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
+env:
+  TEMPLATE_REF: main
+
 jobs:
   test:
     name: ${{ vars.CI_IMAGE }}
     runs-on: ubuntu-latest
     container:
       image: ${{ vars.CI_IMAGE }}
-    env:
-      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
 
     steps:
       - name: Checkout project ⬇️
         uses: actions/checkout@v4
 
-      - name: Checkout Repo Templates      
+      - name: Checkout templates repo ⬇️  
         uses: actions/checkout@v4
         with:
-          ref: main
+          ref: ${{env.TEMPLATE_REF}}
           repository: boehringer-ingelheim/dv.templates
```

Comment:
variable for repo name pls; you can leave the ref var as well

Reply:
done